### ISSUE TITLE
Problem with using ProxylessNAS in example from nni guide

### ISSUE THREAD

**Issue by anadb** [microsoft/nni#2496]
Hello!
I started with the example provided in https://nni.readthedocs.io/en/latest/NAS/NasGuide.html and chose ProxylessNAS as trainer. However I have an error in mutator.py
`object of type 'InputChoice' has no len()`
When I removed all stuff with InputChoice there is a dimension mismatch appeared in first linear layer.

```
class Net(nn.Module):
	def __init__(self):
		super(Net, self).__init__()
		self.conv1 = mutables.LayerChoice([
			nn.Conv2d(1, 32, 3, 1),
			nn.Conv2d(1, 32, 5, 3)
		])
		self.conv2 = nn.Conv2d(32, 64, 3, 1)
		self.dropout1 = nn.Dropout2d(0.25)
		self.dropout2 = nn.Dropout2d(0.5)
		self.fc1 = nn.Linear(9216, 128)
		self.fc2 = nn.Linear(128, 10)

	def forward(self, x):
		x = self.conv1(x)
		x = F.relu(x)
		x = self.conv2(x)
		x = F.relu(x)
		x = F.max_pool2d(x, 2)
		x = self.dropout1(x)
		x = torch.flatten(x, 1)
		x = self.fc1(x)
		x = F.relu(x)
		x = self.dropout2(x)
		x = self.fc2(x)
		output = F.log_softmax(x, dim=1)
		return output
```

**Comment by liuzhe-lz** [comment]
As far as I know ProxylessNAS does not support `InputChoice`. We will fix the doc.
The shape of `fc1` is affected by input data. You can print the shape and change `9216` to correct value.

**Comment by anadb** [comment]
But for the different LayerChoise there will be different tensor size, ho it is possible to initialize LinearLayer dynamically depending on chosen layer?

**Comment by liuzhe-lz** [comment]
Generally `LayerChoice` should only contain layers with identical output shape. This is not about shape inference. `LayerChoice` must have constant output shape as long as the following layer is trainable.
To satisfy this constraint, first you can use padding anywhere to make the image size constant. Then if the channel number varies, you can group each candidate layer with an extra reshaping op, typically an 1x1 conv layer. Besides standard layers you can use custom modules in `LayerChoice` as well.

**Comment by anadb** [comment]
Thank you that's answering my question
